barman/config.py

```diff
@@ -710,6 +729,9 @@ class ServerConfig(BaseConfig):
         "archiver_batch_size": int,
         "autogenerate_manifest": parse_boolean,
         "aws_await_snapshots_timeout": int,
+        "aws_snapshot_lock_duration": int,
+        "aws_snapshot_lock_cool_off_period": int,
+        "aws_snapshot_lock_expiration_date": parse_datetime,
```

Comment:
We could somehow leverage of the utils functions that you created for validating the CLI arguments of `barman-cloud`, to validate all the new configuration options (including the lock mode).